src/Development/IDE/Import/FindImports.hs

```diff
@@ -68,6 +68,12 @@ locateModule
     -> m (Either [FileDiagnostic] Import)
 locateModule dflags exts doesExist modName mbPkgName isSource = do
   case mbPkgName of
+    -- "this" means that we should only look in the current package
+    Just "this" -> do
```

Comment:
Maybe just add a `_pkgName /= "this"` case to the first case and turn the `Nothing` case into a wildcard. Then we don’t need to duplicate the code.

Reply:
Something like that might work, but the code is not actually the same for the `"this"` case and `Nothing` case because we don't even look in the package DB if there's a `"this"`.

Reply:
Ah very good point, probably reasonable to keep it that way for now then.